### ISSUE TITLE
Doesn't work on Windows

### ISSUE THREAD

**Issue by kjk** [danvk/webdiff#188]
On Windows 11 I've installed Python 3.1.2 via Microsoft App Store, installed webdiff via pip and ran `git webdiff`

The app starts but doesn't work and shows this in console:

```
PS C:\Users\kjk\src\edna> git webdiff
Serving diffs on http://localhost:56823
Close the browser tab or hit Ctrl-C when you're done.
 * Serving Flask app 'webdiff.app'
 * Debug mode: off
ERROR:webdiff.app:Exception on /thick/0 [GET]
Traceback (most recent call last):
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\webdiff\app.py", line 207, in thick_diff
    return jsonify(diff.get_thick_dict(DIFF[idx]))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\webdiff\diff.py", line 75, in get_thick_dict
    d.update({'is_image_diff': is_image_diff(diff), 'no_changes': no_changes(diff)})
                                                                  ^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\webdiff\diff.py", line 103, in no_changes
    return util.are_files_identical(diff.a_path, diff.b_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kjk\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\webdiff\util.py", line 43, in are_files_identical
    if os.path.getsize(path1) != os.path.getsize(path2):
       ^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen genericpath>", line 62, in getsize
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: '"C:\\\\Users\\\\kjk\\\\AppData\\\\Local\\\\Temp/git-difftool.a19980/left/src/components/TopNav.svelte"'
```

The issue is: `C:\\\\Users\\\\kjk\\\\` is not a valid path. Somewhere `c:\Users` got transformed into `c:\\\\Users` (I'm not sure if it's `c:\\Users` and escaped for display or is it really `c:\\\\Users`).

This is in powershell with git 2.45.0 installed via winget.

**Comment by danvk** [comment]
Thanks for the report. I don't have a Windows machine, so this might be a bit hard for me to track down. The double-escaping is happening somewhere before the server spins up, so we need some more logs. Are you able to run with then environment variable `DEBUG` set? On my machine I get lots more debug info if I run `DEBUG=1 git webdiff`.

I assume you're on webdiff 1.0.1? You could try updating to 1.1.0, though I doubt it will fix this issue.

Also I'm curious if you can run `webdiff dir1 dir2` to diff two directories? That would at least help pin down where the problem is.

**Comment by kjk** [comment]
Sadly, with DEBUG=1 (or any value) I get:
```
> git webdiff
git webdiff invoked as: ['C:\\Users\\kjk\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python312\\Scripts\\git-webdiff']
error: cannot spawn .\test.sh: No such file or directory
warning: failed: -1
```

**Comment by danvk** [comment]
Try:

    DEBUG=1 git difftool -d -x webdiff

**Comment by kjk** [comment]
Doesn't seem to make a difference.

I did: `$Env:DEBUG=1` because that's powershell to set env variable DEBUG to 1.

The I get:
```
PS C:\Users\kjk\src\tools.arslexis.io> git difftool -d -x webdiff
Serving diffs on http://localhost:64662
Close the browser tab or hit Ctrl-C when you're done.
 * Serving Flask app 'webdiff.app'
 * Debug mode: off
ERROR:webdiff.app:Exception on /thick/0 [GET]
Traceback (most recent call last):
```

I omitted the rest because it's the same but you can see it explicitly says: debug mode: off.

**Comment by danvk** [comment]
I think updating to webdiff==1.1.0 will make the DEBUG environment variable work.

**Comment by kjk** [comment]
If you tell me where and what, I can patch the code on my machine to add debug logs and what not.

**Comment by kjk** [comment]
Updated to 1.1.0, no change

**Comment by danvk** [comment]
Could you try applying this patch (either manually or with git)? I have some hope that the `splitlines` change will do something useful, but the logging will be helpful either way:

https://github.com/danvk/webdiff/commit/96c20ed801b7c2598dc1a1b0a4afb20c9c0a646c

**Comment by kjk** [comment]
```
PS C:\Users\kjk\src\tools.arslexis.io> git difftool -d -x webdiff
RawDiffLine(src_mode='100644', dst_mode='100644', src_sha='0000000', dst_sha='0000000', status='M', path='"C:\\\\Users\\\\kjk\\\\AppData\\\\Local\\\\Temp/git-difftool.a18956/left/scratch.txt"', score=None, dst_path=None)
"C:\\Users\\kjk\\AppData\\Local\\Temp/git-difftool.a18956/left/scratch.txt"
C:\Users\kjk\AppData\Local\Temp/git-difftool.a18956/left/
C:\Users\kjk\AppData\Local\Temp/git-difftool.a18956/right/
C:\Users\kjk\AppData\Local\Temp/git-difftool.a18956/right/..\..\..\..\..\src\tools.arslexis.io\"C:\Users\kjk\AppData\Local\Temp\git-difftool.a18956\left\scratch.txt"
Serving diffs on http://localhost:51067
Close the browser tab or hit Ctrl-C when you're done.
```

**Comment by kjk** [comment]
I've also printed the whole diff:
```
:100644 100644 0000000 0000000 M        "C:\\Users\\kjk\\AppData\\Local\\Temp/git-difftool.a02112/left/scratch.txt"
```

So first, the path need to un-quote `"` and then replace "\\" with "\", potentially only on Windows

**Comment by kjk** [comment]
Actually, this is just generic escaping i.e. `"` is needed to handle paths with spaces and they decided to escape certain chars with `\` so it turns a newline character in \n, tab into \t etc. and therefore also escape the escape char which turns `\` into `\\`

**Comment by danvk** [comment]
I think we're running into this behavior:

```
$ git help diff
...
       -z
           When --raw, --numstat, --name-only or --name-status has been given, do not munge pathnames and use NULs as output field terminators.

           Without this option, pathnames with "unusual" characters are quoted as explained for the configuration variable core.quotePath (see git-config(1)).
```

So the trick is to pass `-z` to `git diff --raw --no-index` and update the parsing to match. I'll push a commit shortly.

Thanks for all the help debugging!

**Comment by kjk** [comment]
Also, here's how I fixed it:
```
    path = path.encode("utf-8").decode("unicode-escape").strip('"')
```
in parse_raw_diff_line

**Comment by kjk** [comment]
That works, thanks.

The output is:
```
PS C:\Users\kjk\src\tools.arslexis.io> git difftool -d -x webdiff
diff:

:100644 100644 0000000 0000000 MC:\Users\kjk\AppData\Local\Temp/git-difftool.a24672/left/scratch.txt


[[':100644 100644 0000000 0000000 M', 'C:\\Users\\kjk\\AppData\\Local\\Temp/git-difftool.a24672/left/scratch.txt']]
```

**Comment by danvk** [comment]
Published 1.1.1 with the fix: https://pypi.org/project/webdiff/

**Comment by kjk** [comment]
Thanks, upgraded to 1.1.1 and it works!
